docs/decisions/adr-0002-cross-database-utils.md

```diff
@@ -0,0 +1,83 @@
+- Cross-database macros serve as a foundation to enable compatibility in other packages
+- Macros to abstract away complex work
+- Useful tests which aren't built into dbt Core
+- A catchall for experiments
+
+The `dbt_utils` package is doing a lot, and it could be split up into more logical chunks. If we pull out each category into a stand-alone package, they can each do their own thing without interfering with one another.
+
+How would this affect users, package maintainers, and adapter maintainers?
+
+## Considered Options
+
+For each category of functionality, there are four main options for its future home:
+* stay in `dbt_utils`
+* move to its own stand-alone package or another existing repository (e.g. [dbt-expectations](https://github.com/calogica/dbt-expectations) in the case of tests, and [dbt-labs-experimental-features](https://github.com/dbt-labs/dbt-labs-experimental-features) in the case of experiments)
+* move to definition in Core, implementation in adapters
+* complete abandonment / deprecation
+
+Since there are four categories and 4 possibilities for destinations, that gives 4^4 = 256 unique options. Rather than enumerate all of them, we'll restrict discussion to a shorter list:
+
+* Migrate cross-db functions from `dbt_utils` to definition in Core, implementation in adapters
+* Split `dbt_utils` into multiple stand-alone packages
+* Keep `dbt_utils` as-is
+
+## Decision Outcome
+
+Chosen option: "Migrate cross-db functions from `dbt_utils` to definition in Core, implementation in adapters", because
+that was the consensus that emerged from the discussion in [dbt-utils #487](https://github.com/dbt-labs/dbt-utils/discussions/487).
+
+Passthroughs will be left behind for migrated macros (so that calls to `dbt_utils.hash` don't suddenly start failing). New cross-database macros can be added in minor and major releases for dbt Core (but not patch releases). End users will retain the ability to use `dispatch` to shim/extend packages to adapters that don't yet support a particular macro.
+
+## Validation
+
+Each moved macro will be validated by leaving a definition in `dbt_utils` and dispatching it to `dbt-core`. Independent continuous integration (CI) testing will exist within `dbt-core`, adapters, and `dbt_utils` using the [new pytest framework](https://docs.getdbt.com/docs/contributing/testing-a-new-adapter).
+
+## Pros and Cons of the Options
+
+### Definition in Core, implementation in adapters
+
+* Good, because common, reusable functionality that differs across databases will work "out of the box"
+* Good, because functionality can subjected to more [rigorous testing](https://docs.getdbt.com/docs/contributing/testing-a-new-adapter)
+* Good, because we hope that many package vendors could drop their dependencies on `dbt_utils` altogether, which makes version resolution easier
+* Good, because it's slightly more convenient to reference the macro as `dateadd` instead of `dbt_utils.dateadd`
```

Comment:
I _think_ I intend to always document them as `dbt.dateadd()` so that it's clear that they're coming from somewhere as opposed to magic macros, so this is maybe a neutral IMO